# HG changeset patch
# User alta88@fixall.com
# Date 1715245786 -7200
# Parent  2e15e5499d9fcd65de6fe7455c614ed341479cea
Fix feed vandalization, Part I - folder/icon css
- lower opacity for folder name, not icon, when paused
- no opacity change for non feed folders
- increase visibility of error icon
- no !important (a poor css solution) for css "content"
- error, busy, new message indicators on the icon are css "content" (for screen readers);
  the icon url is not - it is "background-image"
- reenable error/busy "content" icons to overlap with new messages indicator
- no "fill" to override a site favicon
- for folders with multiple subscriptions, use a favicon (of first sub url found)

Authored by Alta88 and contributed to the Betterbird Project.
See https://github.com/Betterbird/thunderbird-patches/blob/main/LICENSE for license details.
All uses require attribution to the Author.
* * *
Update based on bug 1895118: Override the user customized folder color for busy or error state.

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -4130,12 +4130,13 @@ class FolderTreeRow extends HTMLLIElemen
   setFolderPropertiesFromFolder(folder) {
     if (folder.server.type != "rss") {
       return;
     }
-    let urls = !folder.isServer ? FeedUtils.getFeedUrlsInFolder(folder) : null;
-    if (urls?.length == 1) {
-      let url = urls[0];
-      this.icon.style = `content: url("page-icon:${url}"); background-image: none;`;
+    let urls = FeedUtils.getFeedUrlsInFolder(folder);
+    if (urls) {
+      this.dataset["isFeedFolder"] = "true";
+    } else {
+      delete this.dataset["isFeedFolder"];
     }
     let props = FeedUtils.getFolderProperties(folder);
     for (let name of ["hasError", "isBusy", "isPaused"]) {
       if (props.includes(name)) {
@@ -4143,8 +4144,14 @@ class FolderTreeRow extends HTMLLIElemen
       } else {
         delete this.dataset[name];
       }
     }
+    if (!urls || props.includes("hasError") || props.includes("isBusy")) {
+      this.icon.style = "";
+    } else {
+      let url = urls[0];
+      this.icon.style = `background-image: url("page-icon:${url}");`;
+    }
   }
 
   /**
    * Update this row's name label to match the new `prettyName` of the server.
diff --git a/mail/themes/shared/mail/about3Pane.css b/mail/themes/shared/mail/about3Pane.css
--- a/mail/themes/shared/mail/about3Pane.css
+++ b/mail/themes/shared/mail/about3Pane.css
@@ -432,13 +432,8 @@ li.context-menu-target:not(.selected) > 
 #folderTree li[data-server-type="rss"] > .container > .icon {
   background-image: var(--icon-rss);
 }
 
-#folderTree li[data-server-type="rss"] > ul > li:not([data-folder-type], [data-is-busy="true"]) .icon {
-  --icon-color: currentColor;
-  fill: var(--icon-color);
-}
-
 #folderTree li[data-folder-type="archive"] > .container > .icon {
   background-image: var(--icon-archive);
   --icon-color: var(--folder-color-archive);
 }
@@ -487,38 +482,37 @@ li.context-menu-target:not(.selected) > 
   background-image: var(--icon-newsletter);
   --icon-color: var(--folder-color-newsletter);
 }
 
-#folderTree li[data-is-paused="true"] > .container > .icon,
-#folderTree li[data-is-paused="true"] > .container > .name,
-#folderTree li[data-is-paused="true"] ul > li > .container > .icon,
-#folderTree li[data-is-paused="true"] ul > li > .container > .name {
+#folderTree li[data-is-feed-folder] > .container > .icon {
+  background-image: var(--folder-pane-folder-rss);
+  background-size: 16px 16px;
+}
+
+#folderTree li[data-is-paused] > .container > .name,
+#folderTree li[data-server-type][data-is-paused] ul > li[data-is-feed-folder] > .container > .name {
   opacity: 0.6;
 }
 
-#folderTree li[data-is-busy="true"] > .container > .icon {
-  content: var(--icon-clock) !important;
-  background-image: none;
-  --icon-color: var(--button-primary-background-color);
+#folderTree li[data-is-busy] > .container > .icon,
+#folderTree li[data-has-error] > .container > .icon {
+  background-image: none !important;
+  --icon-color: var(--color-amber-30) !important;
+}
+
+#folderTree li[data-is-busy="true"] > .container > .icon::before {
+  content: var(--icon-clock);
 }
 
 @media (prefers-reduced-motion: no-preference) {
-  #folderTree li[data-is-busy="true"] > .container > .icon {
-    content: var(--icon-loading) !important;
-    animation: activity-indicator-3pane 1.05s steps(30) infinite;
-    object-fit: cover;
-    object-position: 0px 0;
-  }
-  @keyframes activity-indicator-3pane {
-    100% { object-position: -480px 0; }
+  #folderTree li[data-is-busy="true"] > .container > .icon::before {
+    content: url("chrome://global/skin/icons/loading.png");
   }
 }
 
-#folderTree li[data-has-error="true"] > .container > .icon {
-  content: var(--icon-warning) !important;
-  background-image: none;
-  fill: var(--color-amber-30);
-  stroke: var(--color-amber-60);
+#folderTree li[data-has-error] > .container > .icon::before {
+  content: var(--icon-warning);
+  fill: var(--color-red-60);
 }
 
 #folderTree li[data-tag-key] > .container > .icon {
   background-image: var(--icon-tag);
